modules/nuxeo-apidoc-webengine/src/main/java/org/nuxeo/apidoc/browse/Distribution.java

```diff
@@ -192,6 +184,13 @@ protected Object show404() {
         return Response.status(404).type(MediaType.TEXT_HTML_TYPE).entity(Resource404.getPageContent()).build();
     }
 
+    protected void commitOrRollbackAndRestartTransaction() {
+        if (TransactionHelper.isTransactionActiveOrMarkedRollback()) {
```

Comment:
The `if` shouldn't be needed, the caller should always know that it's in a transactional context. (This is a general issue in too much of our code, the no-transaction should be extremely rare and known to the context.)